wfc_dylib/src/lib.rs

```diff
@@ -426,6 +445,10 @@ pub enum WfcObserveResult {
 /// deterministic state or [`WfcObserveResult::Contradiction`] if the
 /// observation made by this function created a world where a slot is occupied
 /// by zero modules.
+/// 
+/// The number of performed observations can be limited if `max_observations`
+/// is set to a non-zero value. For zero the number of observations remains
+/// unlimited.
```

Comment:
While nice, this comment does not actually reflect what is happening. Currently it does at least one observation always. See the main review comment for ideas how to better structure this.